File: tree.py
import sys

def checkArgs():
    """A function to verify command line argument input and report errors"""
    size = 3
    if len(sys.argv) < 3:
        print("Size not specified. Default size is 3.")
    elif len(sys.argv) > 3: # need to stop here
        print("Too many arguments")
        print("Usage: py tree -height [TREE_SIZE]")
    else:
        try:
            size = int(sys.argv[2])
        except ValueError as e:
            print(e, "TREE_SIZE must be a integer")
    return size

def printTree(size: int):
    """A function to print a correctly formatted Christmas tree of the specified size"""
    print("*".center(((size * 2) + 1)))
    midSpace = 1
    for sect in reversed(range(size)):
        print("/".rjust(sect + 1), "\\".rjust(midSpace))
        midSpace += 2
    print("-".center(((size * 2) + 1), "-"))
    print("#".center(((size * 2) + 1)))

def main():
    """A driver function for using checkArgs and printTree to build a Christmas tree"""
    size = 0
    size = checkArgs()
    printTree(size)

if __name__ == "__main__":
    main()

File: tree.go
package main

import (
	"os"
	"fmt"
	"strconv"
	"strings"
)

func checkArgs() int {
	size := 3
	if len(os.Args) < 3 {
		println("Size not specified. Defaulting to size 3.")
	} else if len(os.Args) > 3 {
		println("Error: Too many arguments")
        println("Usage: go run tree.go -height [TREE_SIZE]")
        println("Exiting...")
        os.Exit(1)
	} else {
		size, err := strconv.Atoi(os.Args[2])
		if err == nil {
			return size
		}
		println("Error: TREE_SIZE must be a integer")
		os.Exit(1)
	}
	return size
}

func printTree(size int) {
	fmt.Printf("%*s\n", size + 1, "*")
	midSpace := 2
	for i := size; i > 0; i-- {
		fmt.Printf("%*s ", i, "/")
		fmt.Printf("%*s", midSpace, "\\\n")
		midSpace += 2
	}
	println(strings.Repeat("-", ((size * 2) + 1)))
	fmt.Printf("%*s\n", size + 1, "#")
}

func main() {
	size := checkArgs()
	printTree(size)
}
